src/common/sizer.cpp

```diff
@@ -968,10 +968,27 @@ wxSize wxSizer::ComputeFittingWindowSize(wxWindow *window)
     return window->ClientToWindowSize(ComputeFittingClientSize(window));
 }
 
+#ifdef __WXGTK3__
+static void FitOnShow(wxShowEvent& event)
+{
+    wxWindow* win = static_cast<wxWindow*>(event.GetEventObject());
+    win->GetSizer()->Fit(win);
```

Comment:
This seems unlikely, but in principle it's possible to delete the window sizer between calling `Fit()` and the window being shown, so I'd check if it's still valid here.